src/csharp/Grpc.Core/CallOptions.cs

```diff
@@ -69,7 +69,7 @@ public Metadata Headers
         /// <summary>
         /// Call deadline.
         /// </summary>
-        public DateTime? Deadline
+        public DateTimeOffset? Deadline
```

Comment:
Note that all the API changes need to be backward compatible compatible (otherwise the change breaks all users and that's not acceptable).  It might be ok to add new fields/extension methods  to allow specifying the deadline in form of DateTimeOffset or a timeout, but we can't break the existing API.